src/components/content/exercises/exercise.tsx

```diff
@@ -44,10 +49,33 @@ export function Exercise({ node }: ExerciseProps) {
 
       {renderToolsAndLicense()}
 
-      {solutionVisible && renderSolutionBox()}
+      {solutionVisible && renderSolution()}
     </Wrapper>
   )
 
+  function renderSolution() {
+    return (
+      <Solution
```

Comment:
You have alos added solution to articleRenderer, so it's probably good to use the articleRenderer here.